torch/csrc/jit/codegen/cuda/ir_internal_nodes.h

```diff
@@ -182,6 +187,46 @@ class TORCH_CUDA_API ReductionOp : public Expr {
   Val* const in_ = nullptr;
 };
 
+/*
+ * Reduction operatoin. Out is first initialized to _init. Then
+ * _reduction_op_type is used to update out as out = reductionOp(out, in).
+ * Output's axes marked as reduction will be reduced to produce an output
+ * tensor. The output tensors size will be the size of all
+ * non-reduction/non-broadcast dimensions.
+ */
+class TORCH_CUDA_API GridReduction : public Expr {
```

Comment:
needs cloning constructor

Reply:
I actually don't think we can even call cloning on this node as it's only used during lowering and we don't have a lowered container that uses the cloner.

Reply:
We should probably start thinking about a more formal lowered container and how we might want to use it.

Reply:
never mind, it will blindly copy everything in the fusion so we do need it in case someone copies the fusion after lowering

Reply:
correct. this is even covered by the copy/move test cases.

I agree that we need to create a specialized lowered IR container though.